pkg/proxy/ipvs/proxier_test.go

```diff
@@ -4044,15 +4044,7 @@ func TestCleanLegacyService(t *testing.T) {
 		fp.ipvs.AddVirtualServer(currentServices[v])
 	}
 
-	fp.netlinkHandle.EnsureDummyDevice(defaultDummyDevice)
-	activeBindAddrs := map[string]bool{"1.1.1.1": true, "2.2.2.2": true, "3.3.3.3": true, "4.4.4.4": true}
-	// This is ipv4-only so ipv6 addresses should be ignored
-	currentBindAddrs := []string{"1.1.1.1", "2.2.2.2", "3.3.3.3", "4.4.4.4", "5.5.5.5", "6.6.6.6", "fd80::1:2:3", "fd80::1:2:4"}
-	for i := range currentBindAddrs {
-		fp.netlinkHandle.EnsureAddressBind(currentBindAddrs[i], defaultDummyDevice)
-	}
```

Comment:
It seems like we're losing a lot of testing here... should we be redoing some of this by testing `EnsureAddressBind` after calling `syncProxyRules`?

Reply:
Checking...

Reply:
> To test this in the current implementation feels like testing that set operations work. That's why I removed them.

Well, we want to be sure that the proxier is constructing the _right_ sets, so it's not necessarily totally superfluous.

Reply:
But in this test both the "bindedAddresses" and currentAddresses were defined in the test-case. So it didn't test the right sets, just the set operation (the home-brewed one).

Reply:
To make a similar test I can break out the code above to a function:
```go
func (proxier *Proxier) removeStaleAddresses(activeBindAddrs,  alreadyBoundAddrs sets.Set[string]) {
	// Remove superfluous addresses from the dummy device
	superfluousAddresses := alreadyBoundAddrs.Difference(activeBindAddrs)
	if superfluousAddresses.Len() > 0 {
		klog.V(2).InfoS("Removing addresses", "interface", defaultDummyDevice, "addresses", superfluousAddresses)
		for adr := range superfluousAddresses {
			if err := proxier.netlinkHandle.UnbindAddress(adr, defaultDummyDevice); err != nil {
				klog.ErrorS(err, "UnbindAddress", "interface", defaultDummyDevice, "address", adr)
			}
		}
	}
}
```
bind the addresses in the faked netlinkHandle, call the function with pre-set address, and check addresses in the faked netlinkHandle afterwards. Only addresses with the same family though.